clang/test/Analysis/MismatchedDeallocator-checker-test.mm

```diff
@@ -14,6 +14,13 @@
 void free(void *);
 void __attribute((ownership_takes(malloc, 1))) my_free(void *);
 
+void __attribute((ownership_returns(malloc1))) *my_malloc1(size_t);
+void __attribute((ownership_takes(malloc1, 1))) my_free1(void *);
+
+void __attribute((ownership_returns(malloc2))) *my_malloc2(size_t);
+void __attribute((ownership_returns(malloc2))) *my_malloc3(size_t);
+void __attribute((ownership_takes(malloc2, 1))) __attribute((ownership_takes(malloc3, 1))) my_free23(void *);
```

Comment:
I'm surprised to see that a function can have two different `ownership_takes` attributes.
Are you sure that this is the case? If so, why is that?

Have you considered adding some documentation to the attributes page, and to the Malloc checker to highlight this attribute?

Reply:
I also find this quite strange, but can't say if there are any real-world examples of function that takes ownership of 2 different classes.

Maybe it would be better to just disallow 2 different classes?

Reply:
To me, it would make more sense to allow at most one of this attribute to be present.

Reply:
Disallowed 2 ownership_takes with different classes